server/src/main/kotlin/suwayomi/tachidesk/global/impl/sync/SyncManager.kt

```diff
@@ -0,0 +1,379 @@
+                val localManga = localMangaMap[compositeKey]
+                when {
+                    // Checks if the manga is in favorites and needs updating or adding
+                    remoteManga.favorite -> {
+                        if (localManga == null || isMangaDifferent(localManga, remoteManga)) {
+                            logger.debug { "Adding to favorites: ${remoteManga.title}" }
+                            favorites.add(remoteManga)
+                        } else {
+                            logger.debug { "Already up-to-date favorite: ${remoteManga.title}" }
+                        }
+                    }
+                    // Handle non-favorites
+                    !remoteManga.favorite -> {
+                        logger.debug { "Adding to non-favorites: ${remoteManga.title}" }
+                        nonFavorites.add(remoteManga)
+                    }
+                }
+            }
+        }
+
+        val minutes = elapsedTimeMillis / 60000
+        val seconds = (elapsedTimeMillis % 60000) / 1000
+        logger.debug { "Filtering completed in ${minutes}m ${seconds}s. Favorites found: ${favorites.size}, Non-favorites found: ${nonFavorites.size}" }
+
+        return Pair(favorites, nonFavorites)
+    }
+
+    private fun updateNonFavorites(nonFavorites: List<BackupManga>) {
+        val localMangaList = getAllMangaFromDB()
+
+        val localMangaMap = localMangaList.associateBy { Triple(it.sourceId.toLong(), it.url, it.title) }
+
+        nonFavorites.forEach { nonFavorite ->
+            val key = Triple(nonFavorite.source, nonFavorite.url, nonFavorite.title)
+            localMangaMap[key]?.let { localManga ->
+                if (localManga.inLibrary != nonFavorite.favorite) {
+                    val updatedManga = localManga.copy(inLibrary = nonFavorite.favorite)
+                    updateManga(updatedManga)
+                }
+            }
+        }
+    }
+
+    private fun updateManga(manga: MangaDataClass) {
+        transaction {
+            MangaTable.update({ MangaTable.id eq manga.id }) {
+                it[MangaTable.url] = manga.url
+                it[MangaTable.title] = manga.title
+                it[MangaTable.initialized] = manga.initialized
+
+                it[MangaTable.artist] = manga.artist
+                it[MangaTable.author] = manga.author
+                it[MangaTable.description] = manga.description
+                it[MangaTable.genre] = manga.genre.joinToString(separator = ", ")
+
+                it[MangaTable.status] = MangaStatus.valueOf(manga.status).value
+                it[MangaTable.thumbnail_url] = manga.thumbnailUrl
+                it[MangaTable.thumbnailUrlLastFetched] = manga.thumbnailUrlLastFetched
+
+                it[MangaTable.inLibrary] = manga.inLibrary
+                it[MangaTable.inLibraryAt] = manga.inLibraryAt
+
+                it[MangaTable.sourceReference] = manga.sourceId.toLongOrNull() ?: 0L
```

Comment:
I would prefer that this errors instead of inserting a broken manga.
```suggestion
                it[MangaTable.sourceReference] = manga.sourceId.toLong()
```